### PR TITLE
Converts to new snippets specs - fixes #778

### LINKED ISSUE #778: Dynamically remove the need for version numbers in the code
That would mean that coders wouldn't have to update all the code whenever a version number changed (that kind of move could be an epic pain).

The idea would be to use 'interpolation.js' to strip the version number. This is the summation of the discussion of the idea:

1. New people looking at the code might say to themselves, 'Wait, the version numbers are gone. Something must be wrong. I'll add them back in again,' or be otherwise uncertain enough to use version numbers when they shouldn't. We've seen similar behavior before. We decided part of the problem with that is that the change is almost invisible - both names make sense on their own and there's no way to track down exactly what's going on. There's nothing to search for in the code.
2. Another option is to just strip the numbers off, ending up with something like `foo_v`. That was rejected because of the same basic reasons.
3. The option we liked best was to strip the version number off, but add something at the start - `i_foo` for 'interpolated' or `s_foo` for snippet. That's still mysterious, but clearly weird and intentional, as well as easy to search for.

So option number 3 is the plan!

### PR DESCRIPTION
**Pull request by knod**
This was supposed to be a WIP, but I guess it never made it.

The main logic changes are in 'interpolation.js'. Other than that, it's just updating snippet names in the code.

### CODE REVIEW

**Comment by jlangevin** [comment]
This is a lot of work.  Nice job!  

I only have one comment, but approve without this, since this is personal preference.  Since were already using a template literal here for just `i_`, I like to leverage that and get rid of the string concatenation altogether. 

```
// src/components/StepBar.js
var newStep = { title: { content: snippets[ `i_` + step.key ] }};
becomes:
var newStep = { title: { content: snippets[ `i_${step.key}` ] }};
```

I think they really start to shine as concatenations get longer.
```
// src/forms/CurrentBenefits.js
labelText          = { snippets[ `i_` + name + `Label` ] }
becomes:
labelText          = { snippets[ `i_${name}Label` ] }
```

File: src/localization/en.js [hunk 1]
```diff
@@ -115,17 +115,12 @@ export default {
...
-        label: `Do you have SNAP?`,
-        hint:  `SNAP provides assistance with buying food`,
-      },
-      next: `Next`,
+      currentBenefits_v1:  `Current Benefits`,
+      selectBenefits_v1:   `Select the benefits you currently receive`,
+      hasSection8Label_v1: `Do you have Section 8 Housing?`,
+      hasSection8Hint_v1:  `Section 8 provides rental housing assistance.`,
+      hasSnapLabel_v1:     `Do you have SNAP?`,
+      hasSnapHint_v1:      `SNAP provides assistance with buying food`,
```
[the remark this answers is not in the document]
Comment (knod) [reply src/localization/en.js]: I'd prefer we flatten it out. I started doing that and then realized that wasn't the purpose of this PR. I ended up not undoing that work, but didn't take it further.

**Comment by knod** [comment]
@ethanbb : Great! What more converting do you see to do? I thought I'd caught all of it, but I'd believe you if you said I'd missed some. It looks like you don't think it's a blocking change for this PR. Do you want to put those in an issue so it gets into the collective pool of knowledge?

@jlangevin : Switching to template literals is something that we've turned down so far, but we can definitely discuss it again and reexamine our conclusions. Feel free to bring it up in Slack or wait till Tuesday, whichever works best for you.

**Comment by ethanbb** [comment]
@knod I'm sorry, I was totally wrong about that. I was looking at how you changed `interpolateSnippets` and I interpreted it as including both methods as an intermediate step between returning just the old and just the new versions. (Was I right about that?) And then my sleep-deprived brain decided that meant that going through and changing all the snippet references wasn't part of this PR, even though I had already looked through those changes.

Looking back at `interpolateSnippets` though, I actually do have an issue. It seems inefficient to calculate the full old and new trees of snippets for each nested object at each level, just to discard one of them when the parent merges with `{ ...named, ...old }`. If we have converted over all the snippets in the code, we should be able to just get rid of the "old" version, right? But if you want to keep it in there for backwards compatibility or something, can we at least get rid of the `langCode` and nested object cases in the old version, i.e. the things that are just going to get discarded when the old merges with the new?

**Comment by knod** [comment]
@ethanbb : Shoot, I just forgot to take that out. It was a temporary measure. Will remove!

**Pull request merged**